python/tchannel/tornado/peer.py

```diff
@@ -422,6 +423,84 @@ def score(self):
             # control randomness.
 
 
+def retry(func):
+    @gen.coroutine
+    @functools.wraps(func)
+    def func_wrapper(self, endpoint, arg2, arg3, headers,
```

Comment:
Because this decorator will be used on a instance method. The first argument by default is `self`